onnxruntime/core/framework/allocatormgr.h

```diff
@@ -33,4 +39,24 @@ struct AllocatorCreationInfo {
 // Valid values can be found in onnxruntime_c_api.h.
 AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info);
 
+// TODO: Only used for TRT and CUDA EP currently, need to add more identifiers to use it across all EPs
+class AllocatorManager {
+  //
+ public:
+  AllocatorManager() = default;
+  void InsertAllocator(AllocatorPtr allocator);
+  void ReplaceAllocator(AllocatorPtr allocator);
+  //Get an allocator with specified device id and MemType. Return nullptr if it doesn't exist
+  AllocatorPtr GetAllocator(int id, OrtMemType mem_type) const;
+
+ private:
+  AllocatorMap allocators_;
+  // to ensure only allocators with unique OrtMemoryInfo are registered in the provider.
+  MemoryInfoSet mem_info_set_;
+  
+  // convenience list of the allocators so GetAllocatorList doesn't have to build a new vector each time
```

Comment:
>GetAllocatorList [](start = 43, length = 16)

Didn't find GetAllocatorList..

Reply:
leave it to next PR

---
In reply to: [564299676](https://github.com/microsoft/onnxruntime/pull/6332#discussion_r564299676) [](ancestors = 564299676)